### ISSUE TITLE
[v3] - Expo Link with asChild breaks `Unistyles C++` state

### ISSUE THREAD

**Issue by jpudysz** [jpudysz/react-native-unistyles#432]
### Description

The issue was reported on Discord.  
When attempting to use the `Link` component from `Expo Router` with the `asChild` flag, it breaks the `C++ style` and throws the `Unistyles not bound!` error.

![image](https://github.com/user-attachments/assets/8652a810-1159-4797-9a26-37d557957338)

Using an array helps prevent the error, but it doesn't apply the styles. This happens because `Unistyle` is -- once again, broken.

To resolve this, we need to implement a fallback mechanism for the `Style is not bound!` error, allowing us to manually look for the `ShadowNode`. 

### Steps to Reproduce

1. Bootstrap Expo Router app
2. Add `Link` with `asChild` prop
3. Move styles to `Pressable`

Observe `Unistyle is not bound!` error.

### Unistyles Version

3.0.0-beta.2

### React Native Version

0.76.0

### Platforms

iOS

### Expo

Yes

**Comment by jpudysz** [comment]
Fixed in #462 

Although I tested it with `expo: 52.0.25` and using following syntax:

```tsx
// works
<Link href="/explore" asChild>
    <Pressable style={styles.button}>
        <Text>
            Explore
        </Text>
    </Pressable>
</Link>
```

```tsx
// works
<Link href="/explore" asChild style={styles.button}>
    <Pressable>
        <Text>
            Explore
        </Text>
    </Pressable>
</Link>
```

Doesn't work:

```tsx
// works
<Link href="/explore" asChild>
    <Pressable style={[styles.button]}>
        <Text>
            Explore
        </Text>
    </Pressable>
</Link>
```

`Link` passes following style:

```
 [{"0": {"alignItems": "center", "backgroundColor": "#00d2d3", "borderRadius": 8, "height": 60, "justifyContent": "center", "padding": 10, "width": "100%"}}]
```

So, it can't be handled by Unistyles.
